File: Another_Web/node_basic/database/d1.js
// mysql 모듈을 이용하여 노드에서 mysql 엑세스, 쿼리 수행
// 1. 모듈 가져오기
const mysql = require('mysql')

// 2. 연결 객체 생성
var connection = mysql.createConnection({
    host     : 'localhost',
    user     : 'root',
    password : '123412345',
    database : 'nodedb'
});
// 아래 코드는 동기식 코드이다
// 그런데, 비번틀리게 설정하고 진행해 보면 코드는 쭉 진행되는데, 
// 나중에 문제가 발생된게 등장한다. -> 타이밍이 맞지 않는다 -> 이런스타일로는
// node기반으로 구성할수 없다
// 3. 연결
connection.connect();
console.log('연결 완료')

// 4. 해제  
connection.end();
console.log('연결 종료')



// connection.query('SELECT 1 + 1 AS solution', function (error, results, fields) {
// if (error) throw error;
// console.log('The solution is: ', results[0].solution);
// });

File: Ml/database/d1.py
import pymysql

# 데이터 베이스 접속은 I/O이므로 예외 상황이 항시 발생할수 있다 -> try~
connection = None
try:
  # 1. 데이터베이스 접속 객체 생성 => 접속 정보 제공 => 접속
  connection = pymysql.connect(host='localhost',
                              user='root',
                              password='12341234',
                              db='python_db',
                              charset='utf8mb4')
                
  # with connection.cursor() as cursor:        
  #     sql = "INSERT INTO `users` (`email`, `password`) VALUES (%s, %s)"
  #     cursor.execute(sql, ('webmaster@python.org', 'very-secret'))    
  # connection.commit()
  # with connection.cursor() as cursor:        
  #     sql = "SELECT `id`, `password` FROM `users` WHERE `email`=%s"
  #     cursor.execute(sql, ('webmaster@python.org',))
  #     result = cursor.fetchone()
  #     print(result)
except Exception as e:
  print('오류발생', e)
finally:
  # 2. 접속 닫기  
  if connection:
    connection.close()